splink/comparison_helpers.py

```diff
@@ -0,0 +1,433 @@
+import re
+
+import altair as alt
+import jellyfish
+import pandas as pd
+import phonetics
+
+
+def jaccard_similarity(str1, str2):
+    set1 = set(str1)
+    set2 = set(str2)
+    return len(set1 & set2) / len(set1 | set2)
+
+
+def comparator_score(str1, str2, decimal_places=2):
+    """Helper function to give the similarity between two strings for
+    the string comparators in splink.
+
+    Examples:
+        ```py
+        import splink.comparison_helpers as ch
+
+        ch.comparator_score("Richard", "iRchard")
+        ```
+    """
+
+    scores = {}
+
+    # Levenshtein distance
+    lev_dist = jellyfish.levenshtein_distance(str1, str2)
+    scores["levenshtein_distance"] = round(lev_dist, decimal_places)
+
+    # Damerau-Levenshtein distance
+    dlev_dist = jellyfish.damerau_levenshtein_distance(str1, str2)
+    scores["damerau_levenshtein_distance"] = round(dlev_dist, decimal_places)
+
+    # Jaro distance
+    jaro_sim = jellyfish.jaro_distance(str1, str2)
+    scores["jaro_similarity"] = round(jaro_sim, decimal_places)
+
+    # Jaro-Winkler distance
+    jw_sim = jellyfish.jaro_winkler(str1, str2)
+    scores["jaro_winkler_similarity"] = round(jw_sim, decimal_places)
+
+    # Jaccard similarity
+    jaccard_sim = jaccard_similarity(str1, str2)
+    scores["jaccard_similarity"] = round(jaccard_sim, decimal_places)
+
+    return scores
+
+
+def distance_match(distance, threshold):
+    if distance <= threshold:
+        return True
+    else:
+        return False
+
+
+def similarity_match(similarity, threshold):
+    if similarity >= threshold:
+        return True
+    else:
+        return False
+
+
+def threshold_match(comparator, score, distance_threshold, similarity_threshold):
+    if re.search("distance", comparator):
+        return distance_match(score, distance_threshold)
+    elif re.search("similarity", comparator):
+        return similarity_match(score, similarity_threshold)
+
+
+def comparator_score_df(list, col1, col2):
+    """Helper function returning a dataframe showing the sting similarity
+    scores and string distances for a list of strings.
+
+    Examples:
+        ```py
+        import splink.comparison_helpers as ch
+
+        ch.comparator_score_df(data, "string1", "string2")
```

Comment:
These examples are rather confusing.

Can you add more info on what constitutes `data`

Reply:
have changed to be more explicitly a list input in parameter naming and examples